homeassistant/components/sensor/recorder.py

```diff
@@ -69,13 +73,44 @@ def compile_statistics(
 
         # Make calculations
         if "max" in wanted_statistics:
-            result[entity_id]["max"] = max(fstates)
+            result[entity_id]["max"] = max(*itertools.islice(zip(*fstates), 1))
         if "min" in wanted_statistics:
-            result[entity_id]["min"] = min(fstates)
+            result[entity_id]["min"] = min(*itertools.islice(zip(*fstates), 1))
 
         # Note: The average calculation will be incorrect for unevenly spaced readings,
         # this needs to be improved by weighting with time between measurements
         if "mean" in wanted_statistics:
-            result[entity_id]["mean"] = statistics.fmean(fstates)
+            result[entity_id]["mean"] = fmean(*itertools.islice(zip(*fstates), 1))
```

Comment:
Seems very inefficient to calculate this 3 times.

Reply:
islice and zip both just set up iterators, I don't think it's a problem.
I agree it doesn't look nice though, will try to improve it.